feat(preview): consume client (#425)
codesandbox/sandpack: website/docs/docs/advanced-usage/components.md

### Additional content
For advanced use cases, children of `<SandpackPreview>` are rendered at the end of the preview container.

> ### Getting client instanceSandpackPreviewRef

Reviewer: ```suggestion
### Getting the client instance with SandpackPreviewRef
```

Not sure if that makes sense, but the title did not compute with me considering that the code snippet uses `SandpackPreviewRef`

Author: ```suggestion
### Getting the client instance from SandpackPreview
```

Reviewer: Reads good!

> You can imperatively retrieve the Sandpack client from a SandpackPreview, consume and interact with the current state of the preview. Check out the [type definitions](/api/react/interfaces/SandpackPreviewRef) for more details.

Reviewer: ```suggestion
You can imperatively retrieve the Sandpack client from a SandpackPreview ref, and also consume or interact with the current state of the preview. Check out the [type definitions](/api/react/interfaces/SandpackPreviewRef) for more details.
```

Based on the title, I adjusted the wording here lightly.

```jsx
import { SandpackPreviewRef, useSandpack, SandpackPreview } from "@codesandbox/sandpack-react"

const SandpackPreviewClient: React.FC = () => {
  const { sandpack } = useSandpack();
  const previewRef = React.useRef<SandpackPreviewRef>();

  React.useEffect(() => {
    const client = previewRef.current?.getClient();
    const clientId = previewRef.current?.clientId;

    if (client && clientId) {
      console.log(client);
      console.log(sandpack.clients[clientId]);
    }
  /**
   * NOTE: In order to make sure that the client will be available
   * use the whole `sandpack` object as a dependencie.
   */
  }, [sandpack]);

  return <SandpackPreview ref={previewRef} />;
};
```

:::note

> Worth mentioning that the SandpackClient will not be instantly available. Sandpack has its own rules to decide what is the "right" moment to initialize an instance from a preview component (take into account properties such as autorun, initMode, and the current client stack priority), which means that it's expected that `getClient` function returns `undefined` and it's a valid state.

Reviewer: ```suggestion
Worth mentioning that the SandpackClient will not be instantly available. Sandpack has its own rules to decide when it'is the "right" moment to initialize an instance from a preview component. (Sandpack will take into account properties such as autorun, initMode, and the current client stack priority) 
This means that it's expected that `getClient` function returns `undefined` which is a valid state.
```
